src/openforms/emails/digest.py

```diff
@@ -541,80 +540,84 @@ def collect_expired_or_near_expiry_referentielijsten_data():
+                == DataSrcOptions.reference_lists
             ):
-                service = config.referentielijsten_services.filter(
+                service = config.reference_lists_services.filter(
                     slug=component["openForms"]["service"]
                 ).first()
 
                 if service:
+                    service_label = service.label
+
                     with build_client(
-                        service, client_factory=ReferentielijstenClient
+                        service, client_factory=ReferenceListsClient
                     ) as client:
                         try:
-                            response = client.get(
-                                "tabellen",
-                                params={"code": component["openForms"]["code"]},
-                            )
-                        except RequestException:
-                            response = []
-
-                        service_label = service.label
-                        tables = list(pagination_helper(client, response.json()))
-
-                        # expired or near expiry tabels
-                        for table in tables:
-                            if (
-                                tabel_eindatum := table.get("einddatumGeldigheid")
-                            ) and check_expired_or_near_expiry(tabel_eindatum):
+                            table = client.get_table(component["openForms"]["code"])
+                        except RequestException as e:
+                            return [
+                                ExpiringReferenceListsService(
+                                    service=service_label,
+                                    tables=[],
+                                    table_items=[],
+                                    exception_message=e.args[0],
+                                )
+                            ]
+
+                        # expired or near expiry tables
+                        if table:
+                            if table.expires_on and (
+                                table.is_expired or table.is_nearly_expired
+                            ):
                                 service_tables_items_mapping[service_label][
                                     "tables"
                                 ].append(
-                                    ExpiredOrNearExpiryReferentielijstenTabel(
-                                        naam=table["naam"],
-                                        eindatum=datetime.fromisoformat(tabel_eindatum),
-                                        expired=datetime.fromisoformat(tabel_eindatum)
-                                        <= timezone.now(),
+                                    ExpiredOrNearExpiryReferenceListsTable(
+                                        name=table.name,
+                                        expires_on=table.expires_on,
+                                        expired=table.is_expired,
```

Comment:
do we need to copy this information in another dataclass? Can we not just pass the `Table` and `TableItem` instances?